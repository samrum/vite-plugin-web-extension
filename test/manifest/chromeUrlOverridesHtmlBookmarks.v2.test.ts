import { getResourceDir, runManifestV2Test } from "./manifestTestUtils";

const resourceDir = getResourceDir("chromeUrlOverridesHtml");

runManifestV2Test("chromeUrlOverridesHtmlBookmarks", () => ({
  chrome_url_overrides: {
    bookmarks: `${resourceDir}/bookmarks.html`,
  },
}));
