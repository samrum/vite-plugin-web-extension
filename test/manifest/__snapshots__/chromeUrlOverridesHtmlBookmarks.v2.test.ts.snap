// Vitest Snapshot v1

exports[`chromeUrlOverridesHtmlBookmarks - Manifest V2 1`] = `
[
  {
    "code": "function log(message) {
  console.log(message);
}
log(\\"chromeUrlOverridesHtml\\");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "/mocked-repo-dir/test/manifest/resources/chromeUrlOverridesHtml/bookmarks.html",
    "fileName": "assets/test/manifest/resources/chromeUrlOverridesHtml/bookmarks.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "/mocked-repo-dir/test/manifest/resources/chromeUrlOverridesHtml/bookmarks.html": {
        "code": null,
        "originalLength": 241,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 0,
      },
      "/mocked-repo-dir/test/manifest/resources/chromeUrlOverridesHtml/chromeUrlOverridesHtml.js": {
        "code": "log(\\"chromeUrlOverridesHtml\\");",
        "originalLength": 67,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 30,
      },
      "/mocked-repo-dir/test/manifest/resources/shared/log.js": {
        "code": "function log(message) {
  console.log(message);
}",
        "originalLength": 65,
        "removedExports": [],
        "renderedExports": [
          "default",
        ],
        "renderedLength": 49,
      },
    },
    "name": "test/manifest/resources/chromeUrlOverridesHtml/bookmarks",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "fileName": "test/manifest/resources/chromeUrlOverridesHtml/bookmarks.html",
    "name": undefined,
    "source": "<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />
    

    <script type=\\"module\\" src=\\"http://example.com/httpModuleScript.js\\"></script>
    <script type=\\"module\\" crossorigin src=\\"/assets/test/manifest/resources/chromeUrlOverridesHtml/bookmarks.js\\"></script>
  </head>
</html>
",
    "type": "asset",
  },
  {
    "fileName": "manifest.json",
    "name": undefined,
    "source": "{
  \\"version\\": \\"1.0.0\\",
  \\"name\\": \\"Manifest Name\\",
  \\"manifest_version\\": 2,
  \\"chrome_url_overrides\\": {
    \\"bookmarks\\": \\"test/manifest/resources/chromeUrlOverridesHtml/bookmarks.html\\"
  }
}",
    "type": "asset",
  },
]
`;
